pkg/stanza/fileconsumer/file.go

```diff
@@ -105,6 +110,7 @@ func (m *Manager) startPoller(ctx context.Context) {
 
 // poll checks all the watched paths for new entries
 func (m *Manager) poll(ctx context.Context) {
+	m.reportMetrics(ctx)
```

Comment:
Would it be possible to increment whenever we open a file and decrement whenever we close one?

Reply:
I think the name can stay the same. IMO users don't need to be exposed to the internal package structure. Sticking with fileconsumer seems specific enough.

Reply:
> Would it be possible to increment whenever we open a file and decrement whenever we close one?

So, files are opened by the `Manager` but are closed by the `Manager` and the `Tracker` as well from what I can see.
Also inside the `Manager`'s operations there are multiple places where files are re-opened and then their readers are closed again like at  https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/pkg/stanza/fileconsumer/file.go#L203 or https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/pkg/stanza/fileconsumer/file.go#L222.

Based on this I thought that it makes sense to simplify and only increment the `openFiles` counter only when we create a fresh reader at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/pkg/stanza/fileconsumer/file.go#L225-L232. 

Incrementing in previous steps and then possibly decrementing after a couple of instrunctions didn't make a lot of sense to me but let me know what you think :).